crypto/keyring/keyring.go

```diff
@@ -842,26 +837,166 @@ func (ks keystore) existsInDb(info Info) (bool, error) {
 	return false, nil
 }
 
-func (ks keystore) writeOfflineKey(name string, pub types.PubKey, algo hd.PubKeyType) (Info, error) {
-	info := newOfflineInfo(name, pub, algo)
-	err := ks.writeInfo(info)
+func (ks keystore) writeOfflineKey(name string, pk types.PubKey) (*Record, error) {
+	k, err := NewOfflineRecord(name, pk)
 	if err != nil {
 		return nil, err
 	}
 
-	return info, nil
+	return k, ks.writeRecord(k)
 }
 
-func (ks keystore) writeMultisigKey(name string, pub types.PubKey) (Info, error) {
-	info, err := NewMultiInfo(name, pub)
+// writeMultisigKey investigate where thisf function is called maybe remove it
+func (ks keystore) writeMultisigKey(name string, pk types.PubKey) (*Record, error) {
+	k, err := NewMultiRecord(name, pk)
 	if err != nil {
 		return nil, err
 	}
-	if err = ks.writeInfo(info); err != nil {
+
+	return k, ks.writeRecord(k)
+}
+
+func (ks keystore) MigrateAll() (bool, error) {
+	var migrated bool
+	keys, err := ks.db.Keys()
+	if err != nil {
+		return migrated, err
+	}
+
+	if len(keys) == 0 {
+		fmt.Println("no keys available for migration")
+		return migrated, nil
+	}
+
+	for _, key := range keys {
+		if strings.Contains(key, addressSuffix) {
+			continue
+		}
+
+		migrated, err = ks.migrate(key)
+		if err != nil {
+			fmt.Printf("migrate err: %q", err)
+			continue
+		}
+	}
+
+	return migrated, nil
+}
+
+// for one key
+func (ks keystore) migrate(key string) (bool, error) {
+	var migrated bool
```

Comment:
this variable is never assigned. Let's remove it and be explicit (returning true / false) in all return statements below.